Server/src/services/dataProcess/JsonFileHandler.ts

```diff
@@ -0,0 +1,200 @@
+        let publisherNameId: number;
+        let publicationID: number;
+        let dataSetDataTypeID: number;
+        let datasetID: number;
+        let unitsID: number;
+        let reprID: number;
+
+        this.parsedFileData = jsonData
+
+        // Create this object after the parsing passes
+        let uploadModel = new DataUploadModel();
+
+        try {
+            referenceTypeID = await uploadModel.insertReferenceType(referenceType); console.log('Received reference ID' + referenceTypeID);
+        } catch (err) {
+            console.log('rejected request for referenceTypeID');
+        }
+
+        referencePublisher = this.parsedFileData.reference.publisher;
+        try {
+            publisherNameId = await uploadModel.insertPublisher(referencePublisher);
+            console.log('Received publisher name ID' + publisherNameId);
+        } catch (err) {
+            console.log('rejected request for inserting publisherNameId')
+        }
+
+
+        referenceAuthors = this.parsedFileData.reference.authors;
+        let allAuthors: Authors[];
+        try {
+            allAuthors = await uploadModel.insertAuthors(referenceAuthors);
+            console.log('reference authors: ' + referenceAuthors);
+        } catch (err) {
+            console.log('reference authors not found....request rejected');
+        }
+
+        referenceTitle = this.parsedFileData.reference.title;
+        referenceDOI = this.parsedFileData.reference.doi;
+        referencePages = this.parsedFileData.reference.pages;
+        referenceYear = this.parsedFileData.reference.year;
+        referenceVolume = this.parsedFileData.reference.volume;
+
+        if (this.parsedFileData.reference.datePublished !== undefined)
+            referenceDatePublished = this.parsedFileData.reference.datePublished;
+        if (this.parsedFileData.reference.dateAccessed !== undefined)
+            referenceDateAccessed = this.parsedFileData.reference.dateAccessed;
+
+        try {
+            publicationID = await uploadModel.insertPublication(referenceTitle, referenceDOI, referencePages, referenceTypeID, publisherNameId, referenceYear, referenceVolume, referenceDatePublished, referenceDateAccessed, allAuthors);
+            console.log('received publicationID ' + publicationID);
+        }
+        catch (err) {
+            console.log('publicationID was not received......rejecting request');
+        }
+
+        material = this.parsedFileData.material;
+        let allMaterials: any[];
+        try {
+            allMaterials = await uploadModel.insertMaterial(material);
+            console.log('received material(s)' + material);
+        } catch (err) {
+            console.log('material(s) not found');
+        }
+
+        category = this.parsedFileData.category;
+        subcategory = this.parsedFileData.subcategory;
+        let categoryIDs = await uploadModel.insertCategories(category, subcategory);
+
+        dataType = this.parsedFileData["data type"];
```

Comment:
i think we can talk with stakeholder he wouldnt mind having data_type with underscore